multiproof.go

```diff
@@ -175,9 +175,6 @@ func CheckMultiProof(transcript *common.Transcript, ipaConf *ipa.IPAConfig, proo
 	// Compute helper_scalar_den. This is 1 / t - z_i
 	helper_scalar_den := make([]fr.Element, common.POLY_DEGREE)
 	for i := 0; i < common.POLY_DEGREE; i++ {
-		if groupedEvals[i].IsZero() {
-			continue
-		}
```

Comment:
Despite this was trying to avoid potentially unneeded work, it isn't correct.

The value `1/(t - z_i)` is required for two things:
1. To compute `g_2_t` (L189). So here it's true that if the `groupedEvalus[i]`, we don't actually need this `1/(t-z_i)`, so that's fine.
2. To compute `E` (L204) (i.e: `(r^i / t - z_i)`). But this calculation is independent of the aggregated evaluation Y at point Z=X. Even if that evaluation is 0, we still need this `1/(t-z_i)` correctly computed.